### DISCUSSION TITLE
Map Maintenance: Handling of `fixme=name` and suspicious combinations of `name=*+noname=yes`

### DISCUSSION THREAD

**Discussion by kmpoppe** [streetcomplete/StreetComplete#5928] [Feedback]
# Situation

There are currently ~160k objects tagged with `fixme=name`.

Understandably, SC currently ignores `fixme` tags completely - IMHO most values of that key would be impossible to survey without deeper investigation.

Yet, `fixme=name` seems like a very straight-forward request: "I (the last mapper) was unable to determine what the name of this feature is, please help out".

# Proposal

I therefore would suggest for SC to work the following way:

a) remove `fixme=name` (if present) when the user answers quests/road_name or quests/place_name with either a real name or `noname=yes` (because otherwise, things like [this](https://www.openstreetmap.org/way/208400787/history/4) (`noname=yes` set, `fixme=name` persited) or [this](https://www.openstreetmap.org/node/1840355375/history/5) (`name` set, `fixme=name` persisted) happen)

b) the aforementioned quests should force a survey of the name when a `fixme=name` is older than one year to get rid of it, because all viable answer paths from there (`name=*`, `noname=yes` or changing the type of road that doesn't get asked for a name again) lead to a situation where the `fixme` is no longer relevant.

c) `FIXME=name` (1.5k objects) could be handled the same way.

# SCEE

I do not think that this option should only be something relevant for SCEE, because the handling is purely under to hood and there's no expert knowledge necessary to answer those already existing quests.

# Feedback appreciated !

**Comment by matkoniecz** [comment]
I am tempted to propose bot edit that would remove `fixme=name` from objects without any name tags but I admit that right now I have not time to do this.

**Reply by kmpoppe** [reply]
I see what you mean. Maybe, semi-automated with MapRoulette would also yield the same result in the end, but as usual, that's just a one-off dataset (if one doesn't want to constantly feed in new object to fix) - whereas SC would allow for a constant maintenance of the map :)

**Reply by matkoniecz** [reply]
from https://taginfo.openstreetmap.org/tags/fixme=name#chronology it looks like it was brought in large parts by automated mass edits and there is potentially organic growth that stopped and got reversed

so hopefully once cleaned they would not return in meaningful numbers

**Reply by kmpoppe** [reply]
I have now checked the history of all 15,459 objects, that have (`fixme|FIXME`=`name|name?|Name|Name?` and either have `name=*` or `noname=yes`) whereof 4,705 (30.44%) had the `fixme` added _before_ `name` or `noname` was set - those might be objects were I will ask the forum and tagging-ml if somebody would oppose an automated edit.

For the other 7 out of 10 objects a manual check would be needed, as the `fixme` was added, _after_ `name` or `noname` was added.

Nevertheless, 145k objects remain with a set `fixme` without a `name` or `noname`. So, if ever those were edited with SC, the fixme _must_ be removed from hereonout (point a) of my list above) - do you agree?

**Reply by kmpoppe** [reply]
In the same way, sollowing @mnalis's suggestion, 13,055 objects have `name=*+noname=yes`, 9,923 (76%) had their name set _after_ noname was set and they should therefore be included in an automated edit to remove `noname`.

Thoughts?

**Reply by mnalis** [reply]
@kmpoppe both automated edits as well as SC change to remove `fixme=name` (and its ilk) when `name` is set sound good ideas to me!

**Reply by kmpoppe** [reply]
Seeing the replies in the [community](https://community.openstreetmap.org/t/automated-edit-for-name-inconsistencies/119419), the list(s) and the wiki discussion I feel like automation is not the way to move forward, but rather going the MapRoulette way where every object is manually checked before changing.

Am I the only one feeling that?

**Comment by mnalis** [comment]
Makes sense to me. a) (and by extension c)) especially sounds like good stewardship from SC side, as there are about [160k of fixme=name](https://taginfo.openstreetmap.org/tags/fixme=name#combinations) tags! 

b) also makes sense to me, to proactively resolve those ambiguous situations. I though to suggest that might be helpful if quest pre-filled existing `name` if present, but that sounds like it would create more confusion than it would help, and also might lead to people incorrectly confirming name which is not correct... So probably better to ask `name` again from scratch..

In addition, name resurvey could also be applied to `name=*` + `noname=yes` combinations (same [13k matches](https://taginfo.openstreetmap.org/keys/noname#combinations)) and get rid of one of those tags...
